src/test/java/io/cdap/plugin/sfmc/source/MarketingCloudInputSplitTest.java

```diff
@@ -0,0 +1,99 @@
+    Assert.assertEquals("Table Name", actualMarketingCloudInputSplit.getTableName());
+  }
+
+  @Test
+  public void testInputSplitWithEmptyTableName() throws IOException, InterruptedException {
+    MarketingCloudInputSplit actualMarketingCloudInputSplit = new MarketingCloudInputSplit();
+    Assert.assertEquals(0L, actualMarketingCloudInputSplit.getLength());
+    Assert.assertNull(actualMarketingCloudInputSplit.getObjectName());
+    Assert.assertNull(actualMarketingCloudInputSplit.getTableName());
+  }
+
+  @Test
+  public void testReadFields() throws IOException {
+    MarketingCloudInputSplit marketingCloudInputSplit = new MarketingCloudInputSplit("open Event", "Table Name");
+    ObjectInputStream objectInputStream = Mockito.mock(ObjectInputStream.class);
+    Mockito.when(objectInputStream.readUTF()).thenReturn("Utf");
+    marketingCloudInputSplit.readFields(objectInputStream);
+    Assert.assertEquals("Utf", marketingCloudInputSplit.getTableName());
+    Assert.assertEquals("Utf", marketingCloudInputSplit.getObjectName());
+  }
+
+  @Test
+  public void testGetLocations() throws IOException, InterruptedException {
+    Assert.assertEquals(String[].class, new MarketingCloudInputSplit("Table Name", "Object Name").getLocations().
+      getClass());
+    Assert.assertEquals(0, (new MarketingCloudInputSplit("Table Name", "Object Name")).getLocations().length);
+  }
+
+  @Test
+  public void testGetTableName() {
+    String expectedValue = "Table";
+    String tableName = "Table";
+    String objectName = "Object Name";
+    MarketingCloudInputSplit marketingCloudInputSplit = new MarketingCloudInputSplit(objectName, tableName);
+    String actualValue = marketingCloudInputSplit.getTableName();
+    Assert.assertEquals(expectedValue, actualValue);
+  }
+
+  @Test
+  public void testGetObjectName() {
+    String tableName = "Table";
+    String expectedValue = "Object Name";
+    String objectName = "Object Name";
+    MarketingCloudInputSplit marketingCloudInputSplit = new MarketingCloudInputSplit(objectName, tableName);
+    String actualValue = marketingCloudInputSplit.getObjectName();
+    Assert.assertFalse(marketingCloudInputSplit.getObjectName().isEmpty());
+    Assert.assertEquals(expectedValue, actualValue);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testWriteWithNullData() throws IOException {
+    DataOutput dataOutput = null;
+    String tableName = "";
+    String objectName = "";
+    MarketingCloudInputSplit marketingCloudInputSplit = new MarketingCloudInputSplit(tableName, objectName);
+    marketingCloudInputSplit.write(dataOutput);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testRead() throws IOException {
+    DataInput dataInput = null;
+    String tableName = "";
+    String objectName = "";
+    MarketingCloudInputSplit marketingCloudInputSplit = new MarketingCloudInputSplit(objectName, tableName);
+    marketingCloudInputSplit.readFields(dataInput);
```

Comment:
what is the purpose of this test?

Reply:
This is negative test case which throws NullPointer Exception. Does it needs to be removed?

Reply:
This is ok if we have a positive test case